packages/web/src/components/date/DatePicker.js

```diff
@@ -94,7 +94,7 @@ class DatePicker extends Component {
 		}
 	};
 
-	formatInputDate = date => new XDate(date).toString('yyyy-MM-dd');
+	formatInputDate = date => new XDate(date).toString('yyyy-M-d');
```

Comment:
Yeah will take care for queryFormat prop.

I did it intentionally if we keep the format as `yyyy-MM-dd` see the below example : 
![datepicker_standarad_format](https://user-images.githubusercontent.com/22376783/46618476-3bb59680-cb3d-11e8-9c96-1b8c77447411.gif)

If we want to edit two digit date, it automatically parses the date after we delete the second digit like : `17` -> `01`

Reply:
The parsing logic should be delegated until the required number of characters are present. For example, for parsing `yyyy-MM-dd` the parsing should be delegated till there are 10 characters. This would also be applicable for other formats as well. For example, if your format is epoch_millis, you ideally don't want the date to be parsed until the characters are atlease a minimum number. Needs careful checks in the parsing logic for a good UX and covering all formats.

Reply:
Sounds good. I like the idea to keep both the input and queryFormat consistent. 👍 

Should we keep the `input` and `queryFormat` consistent or keep the standard date in `input`? cc @siddharthlatest @metagrover

Reply:
For better UX, I think keeping the input as standard date would be better.

Thoughts @divyanshu013 @siddharthlatest?

Reply:
Okay then, lets keep it standard `yyyy-MM-dd` in the input for all formats. If later a need arises, more parsing options can be added.

Reply:
Not possible to check number of characters in `formatInputDate` as we are not getting the data entered in `Input` instead we get the full Date format